test/embedding/embedtest.cc

```diff
@@ -125,7 +142,13 @@ int RunNodeInstance(MultiIsolatePlatform* platform,
 
     FILE* fp = fopen((snapshot_arg_it + 1)->c_str(), "w");
     assert(fp != nullptr);
-    snapshot->ToFile(fp);
+    if (snapshot_as_file_it != args.end()) {
+      snapshot->ToFile(fp);
+    } else {
+      const std::vector<char> vec = snapshot->ToBlob();
+      size_t written = fwrite(vec.data(), vec.size(), 1, fp);
+      assert(written == 1);
+    }
     fclose(fp);
```

Comment:
I might be entirely wrong about this, but my understanding of `fwrite` so far has been that its return value only indicates how many objects have been written to the (buffered) stream, regardless of whether they've been written to the underlying physical storage. Isn't it necessary to check the result of `fflush()` or `fclose()` to ensure that any buffered data is written to the storage medium?

Reply:
Ah, sorry, I didn't realize I was referring to a test file. I think it's also true for non-test occurrences such as the same call to `ToFile()` in `node.cc`. (I can't seem to get the link to work, sorry).

Maybe add `CHECK_EQ(fflush(out), 0)` to `ToFile()` since it already aborts on certain I/O errors, so that the caller does not have to worry about it? Then it should be reasonably safe to ignore the return value of `fclose()` :slightly_smiling_face: 

> Wouldn’t that apply to https://github.com/nodejs/node/pull/46497 as well then?

I didn't look at what exactly `FromBlob()` is doing for that PR and just moved the `fclose()` call around, but it seems to me that `FromBlob()` cannot run into similar problems because it first determines the size of the entire file and then checks whether it successfully read the number of bytes. Buffering would happen after actual I/O, so if `FromBlob()` succeeds, no relevant I/O error has occurred. (When writing, buffering happens before actual I/O.)

Reply:
Yeah, that makes sense to me. I’ll follow up with a PR to do that.

Reply:
PR: https://github.com/nodejs/node/pull/46531